src/outcome/utils/transactions.py

```diff
@@ -0,0 +1,116 @@
+"""Transactions handler.
+
+Usage examples:
+
+- Default usage auto-applies all added operations:
+```
+with transaction() as ops:
+    op = Operation(apply_fn=your_apply_fn, rollback_fn=your_rollback_fn)
+    ops.add(op)
+```
+
+- We can be disable auto-apply if necessary:
+```
+op = None
+with transaction(auto_apply=False) as ops:
+    op = Operation(apply_fn=your_apply_fn, rollback_fn=your_rollback_fn)
+    ops.add(create_user)
+    op.apply()
+```
+
+"""
+
+from contextlib import contextmanager
+from enum import Enum
+
+from transaction import Transaction, TransactionManager
```

Comment:
@d-tw do we have to do something special as the import will be optional?

Reply:
You can always make it a bit more dev-friendly:

```suggestion
try:
    from transaction import Transaction, TransactionManager
except ImportError as ex:
    raise Exception('You need to install the 'transaction' extra to use the transaction features') 
```